### PR TITLE
Dynamodb with dynamoose

### PR DESCRIPTION
**Pull request by matiasirpac**
Added dynamoose and three layer architecture

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -3,9 +3,24 @@ import { AppController } from './app.controller';
...
+        accessKeyId: process.env.AWS_ID,
+        secretAccessKey: process.env.AWS_SECRET,
+        region: process.env.AWS_REGION,
+      },
+      local: Boolean(process.env.AWS_LOCAL),
+      logger: false,
+    }),
+    UserModule,
+    TaskModule,
+  ],
```
Comment (davidgk) [inline src/app.module.ts]: I would extract this into another file inside of config

File: package.json [hunk 2]
```diff
@@ -6,11 +6,12 @@
   "private": true,
   "license": "UNLICENSED",
   "scripts": {
+    "docker:up": "docker-compose --env-file .env.dev up -d",
```
Comment (davidgk) [inline package.json]: good one!  would add --no-recreate, to do not recreate the containers

File: .example.env [hunk 3]
```diff
@@ -11,6 +11,8 @@ AWS_REGION=us-east-1
 AWS_ENDPOINT=http://localhost:8000
 AWS_DYNAMO_LOCAL_PORT=8000
 AWS_DYNAMO_ADMIN_PORT=8001
+AWS_LOCAL=false
```
Comment (davidgk) [inline .example.env]: within .env.dev with this value, and for prod with true ? correct ?

File: package.json (hunk 2 above)
Comment (davidgk) [reply package.json]: Also I'd add stop.

File: src/app.module.ts [hunk 4]
```diff
@@ -3,9 +3,24 @@ import { AppController } from './app.controller';
...
+  imports: [
+    ConfigProjModule,
+    DynamooseModule.forRoot({
+      aws: {
+        accessKeyId: process.env.AWS_ID,
+        secretAccessKey: process.env.AWS_SECRET,
+        region: process.env.AWS_REGION,
+      },
+      local: Boolean(process.env.AWS_LOCAL),
+      logger: false,
```
Comment (davidgk) [inline src/app.module.ts]: for development purpose.. wouldn't be better for dev environment logger in true ? or it doesn't make sense ?

**Review by davidgk — APPROVED** [review]
I leave you a couple of comments that might worth to add them before merge.

File: src/app.module.ts (hunk 4 above)
Comment (matiasirpac) [reply src/app.module.ts]: Yes, i agree!

File: .example.env (hunk 3 above)
Comment (matiasirpac) [reply .example.env]: Oh! for prod is false, and dev is true.

**Review by davidgk — APPROVED** [review]

**Pull request merged**